src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -4744,6 +4744,25 @@ void addWarning(String warningString) {
         }
     }
 
+    // Any changes to SQLWarnings should be synchronized.
+    void addWarning(SQLWarning sqlWarning) {
+        warningSynchronization.lock();
+        try {
+            if (null == sqlWarnings) {
+                sqlWarnings = sqlWarning;
+            } else {
+                sqlWarnings.setNextWarning(sqlWarning);
+            }
+        } finally {
+            warningSynchronization.unlock();
+        }
+    }
+
+    // Any changes to SQLWarnings should be synchronized.
+    void addWarning(ISQLServerMessage sqlServerMessage) {
```

Comment:
since we now have 2 of these, add comment on how they differ and how they are used

Reply:
done